posthog-react-native/src/PostHogProvider.tsx

```diff
@@ -51,6 +48,31 @@ export const PostHogProvider = ({
   const captureLifecycle =
     !captureNone && posthog && (captureAll || (autocaptureOptions?.captureLifecycleEvents ?? true)) // Default to true if not set
 
+  // Resolve async client to concrete client
+  useEffect(() => {
+    if (!posthog && client) {
+      if (client instanceof Promise) {
+        client.then(setPosthog)
+      } else {
+        setPosthog(client)
+      }
+    }
+  }, [client])
+
+  useEffect(() => {
+    if (apiKey && !posthog) {
+      PostHog.initAsync(apiKey, options).then(setPosthog)
+    }
+  }, [apiKey])
```

Comment:
I think there is a race condition between initializing the client from a given apiKey and an async initialization. Generally I think one would pass either an apiKey or a client, right? Would be good to add a warning for accidentally passing both.

Reply:
That's a good point! I will do exactly that